README.md

```diff
@@ -128,6 +128,42 @@ SolidusStripe.configure do |config|
 end
 ```
 
+## Implementation
+
+### Payment state-machine vs. PaymentIntent statuses
+
+When compared to the Payment state machine, Stripe payment intents have different set of states and transitions.
+The most important difference is that on Stripe a failure is not a final state, rather just a way to start over.
+
+In order to map these concepts SolidusStripe will match states in a slightly unexpected way, as shown below.
+
+Reference: https://stripe.com/docs/payments/intents?intent=payment
+
+![image](https://user-images.githubusercontent.com/1051/217322027-f49081f5-0795-49f4-994e-285a9de5347c.png)
+
+### ⚠️ Warning: Authorization happens before the order is completed
+
+This setup implies that after the payment information are submitted to Stripe the payment is authorized, although
+the order is not yet completed. This can become an issue if a customers abandons the checkout at the `confirm`
+step and no action is taken to free up the money on the backend.
+
+In order to mitigate this issue we suggest to adapt the frontend merging the confirm and payment steps. Embedding
+the agreement to the terms of service and an order summary in the payment step and apply this patch:
+
+```patch
+--- a/templates/app/controllers/checkouts_controller.rb
++++ b/templates/app/controllers/checkouts_controller.rb
+@@ -48,6 +48,8 @@ def redirect_on_failure
+   end
+
+   def transition_forward
++    @order.next if @order.has_checkout_step?("payment") && @order.payment?
```

Comment:
I tried a similar hack but in `solidus-core` itself (calling `.next!` after the transition to `:confirm`), and there were a bunch of errors:

```
rspec ./spec/models/spree/promotion_code_spec.rb:237 # Spree::PromotionCode completing multiple orders with the same code resets the state of the order
rspec ./spec/models/spree/promotion_code_spec.rb:231 # Spree::PromotionCode completing multiple orders with the same code increases the total to remove the promo
rspec ./spec/models/spree/promotion_code_spec.rb:225 # Spree::PromotionCode completing multiple orders with the same code adjusts the promo_total
rspec ./spec/models/spree/promotion_code_spec.rb:219 # Spree::PromotionCode completing multiple orders with the same code makes the promotion ineligible
rspec ./spec/models/spree/reimbursement_spec.rb:112 # Spree::Reimbursement#perform! refunds the total amount
rspec ./spec/models/spree/reimbursement_spec.rb:117 # Spree::Reimbursement#perform! creates a refund
rspec ./spec/models/spree/reimbursement_spec.rb:172 # Spree::Reimbursement#perform! triggers the reimbursement mailer to be sent via subscribed event
rspec ./spec/models/spree/reimbursement_spec.rb:142 # Spree::Reimbursement#perform! with included tax saves the included tax and refunds the total
rspec ./spec/models/spree/reimbursement_spec.rb:157 # Spree::Reimbursement#perform! when reimbursement cannot be fully performed does not send a reimbursement email and raises IncompleteReimbursement error
rspec ./spec/models/spree/reimbursement_spec.rb:127 # Spree::Reimbursement#perform! with additional tax saves the additional tax and refunds the total
rspec ./spec/models/spree/reimbursement_spec.rb:166 # Spree::Reimbursement#perform! when exchange is required generates an exchange shipment for the order for the exchange items
rspec ./spec/models/spree/order/checkout_spec.rb:417 # Spree::Order with default state machine from payment with confirmation required transitions to confirm
rspec ./spec/models/spree/order/checkout_spec.rb:430 # Spree::Order with default state machine from payment when payment is not required does not call process payments
rspec ./spec/models/spree/order/checkout_spec.rb:291 # Spree::Order with default state machine from delivery attempts to apply free shipping promotions
rspec ./spec/models/spree/order/checkout_spec.rb:323 # Spree::Order with default state machine from delivery without payment required transitions to complete
rspec ./spec/models/spree/order/checkout_spec.rb:358 # Spree::Order with default state machine from delivery correctly determining payment required based on shipping information with a shipment that is free skips payment, transitions to confirm
```

I didn't look into the details, and probably at least some failures are because of the test setup itself.

Reply:
Not saying to remove the warning from the README, though.